sample_consensus/include/pcl/sample_consensus/impl/sac_model_cylinder.hpp

```diff
@@ -342,7 +342,7 @@ pcl::SampleConsensusModelCylinder<PointT, PointNT>::projectPoints (
       Eigen::Vector4f p (input_->points[inlier].x,
                          input_->points[inlier].y,
                          input_->points[inlier].z,
-                         1);
+                         0);
```

Comment:
Let's refactor the if-else branches a bit.

Move out L339-L357 and L372-L387 outside the if-else branch. The loop at L340 is same as L373 (just different semantics) so use the semantics of L373.

Resolve the issue just like you did here (manually copy the {x,y,z,0} into Vector4f)